pkg/scheduler/framework/plugins/noderesources/requested_to_capacity_ratio.go

```diff
@@ -53,7 +49,7 @@ type functionShapePoint struct {
 
 // NewRequestedToCapacityRatio initializes a new plugin and returns it.
 func NewRequestedToCapacityRatio(plArgs *runtime.Unknown, handle framework.FrameworkHandle) (framework.Plugin, error) {
-	args := &config.RequestedToCapacityRatioArguments{}
+	args := &schedulerv1alpha2.RequestedToCapacityRatioArgs{}
 	if err := framework.DecodeInto(plArgs, args); err != nil {
```

Comment:
this was previously decoding into an internal type with no json tags?

Reply:
are there any tests exercising loading this config?

Reply:
oh... the "internal" type defined json tags

Reply:
check what happens with the following args in 1.18 for this plugin:

```yaml
shape: []{{"Utilization": 1, "Score": 1}, {"utilization": 2, "score": 2}}
resources: []{{"Name":"a","Weight":1},{"Name":"b","Weight":2}}
```

Reply:
True. There is nothing exercising these E2E. It is non-trivial given how buried the types are.

I think the json decoder would simply ignore casing. @pancernik, could you verify this?

Reply:
We can add the test in `requested_to_capacity_ratio_test.go`, as there is no other location that would go through the decoding. I'll add the test in the compatibility tests once we move the decoding to the component config files.

Reply:
Surprisingly, Go's [JSON encoding](https://golang.org/pkg/encoding/json/#Unmarshal):
> matches incoming object keys to the keys used by Marshal (either the struct field name or its 
tag), preferring an exact match but also accepting a case-insensitive match

I did confirm this when working on this PR.  Wouldn't adding this test add some confusion to the future readers? This test would actually test a documented behaviour of the language stdlib. What do you think @alculquicondor @liggitt ?

Reply:
yes, please add the test and I'll move it to compatibility tests when we are ready.

But leave the types that didn't have json tags without them in v1alpha2 with a comment to fix them in v1beta1.

Reply:
The stdlib behavior is problematic, and API type decoding is explicitly case-sensitive. We need the test to make sure we don't break compatibility with v1alpha2 configurations.

Reply:
The test was added and the types were copied to v1alpha2 without json tags from the second level.